core/commands/refs.go

```diff
@@ -135,7 +135,7 @@ Note: list all refs recursively with -r.
 					return nil, errors.New(obj.Err)
 				}
 
-				return strings.NewReader(obj.Ref), nil
```

Comment:
there are newlines, just not in the string inside the object. As far as delimiting the json output, that doesnt happen here. We would have to make the change inside the `commands/http` lib. I can do that in this PR if you like.